c/tskit/tables.c

```diff
@@ -7118,105 +7118,95 @@ ancestor_mapper_run(ancestor_mapper_t *self)
  * IBD Result
  *************************/
 
+typedef struct {
+    tsk_segment_t *head;
+    tsk_segment_t *tail;
+} tsk_segment_list_t;
+
+static int64_t
+pair_to_integer(tsk_id_t a, tsk_id_t b, tsk_size_t N)
+{
+    tsk_id_t tmp;
+    if (a > b) {
+        tmp = a;
+        a = b;
+        b = tmp;
+    }
+    return ((int64_t) a) * (int64_t) N + (int64_t) b;
```

Comment:
If I understand what is going on here correctly and the result needs to be unique for each pair of nodes then this won't work for 64bit IDs? Just checking I understand that this isn't future proof and it is worth documenting.

Reply:
Hah, good point. I've updated with a discussion and a bug_assert to make sure we can't overflow. If we ever need to deal with tables this big, then I'm sure we'll be hitting a bunch of other problems too :wink:

Reply:
Great! I know it seems ridiculous as we're far away from node tables that big.